kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/main/java/org/kie/pmml/compiler/commons/utils/DerivedFieldFunctionUtils.java

```
@@ -0,0 +1,247 @@
+import org.dmg.pmml.NormContinuous;
+import org.dmg.pmml.NormDiscrete;
+import org.dmg.pmml.TextIndex;
+import org.kie.pmml.commons.model.enums.DATA_TYPE;
+import org.kie.pmml.commons.model.tuples.KiePMMLNameValue;
+
+import static com.github.javaparser.StaticJavaParser.parseClassOrInterfaceType;
+import static org.kie.pmml.compiler.commons.utils.CommonCodegenUtils.OPTIONAL_FILTERED_KIEPMMLNAMEVALUE_NAME;
+import static org.kie.pmml.compiler.commons.utils.CommonCodegenUtils.getFilteredKiePMMLNameValueExpression;
+import static org.kie.pmml.compiler.commons.utils.CommonCodegenUtils.getMethodDeclaration;
+import static org.kie.pmml.compiler.commons.utils.CommonCodegenUtils.getTypedClassOrInterfaceType;
+
+/**
+ * Class meant to provide <i>helper</i> methods to retrieve <code>Function</code> code-generators
+ * out of <code>DerivedField</code>s
+ */
+public class DerivedFieldFunctionUtils {
+
+    static final String KIEPMMLNAMEVALUE_LIST_PARAM = "param1"; // it is the first parameter
+
+    private DerivedFieldFunctionUtils() {
+        // Avoid instantiation
+    }
+
+    static Map<String, MethodDeclaration> getDerivedFieldsMethodMap(final List<DerivedField> derivedFields, final AtomicInteger arityCounter) {
+        Map<String, MethodDeclaration> toReturn = new HashMap<>();
+        derivedFields.forEach(derivedField ->
+                                      toReturn.put(derivedField.getName().getValue(),
+                                                   getDerivedFieldMethodDeclaration(derivedField, arityCounter)));
+        return toReturn;
+    }
+
+    static MethodDeclaration getDerivedFieldMethodDeclaration(final DerivedField derivedField, final AtomicInteger arityCounter) {
+        final Expression expression = derivedField.getExpression();
+        if (expression != null) {
+            return getExpressionMethodDeclaration(expression, arityCounter);
+        } else {
+            // TODO {gcardosi}
+            return new MethodDeclaration();
```

If this is a feature not supported yet I would prefer to raise an exception or at least log error instead of silently skip it. Wdyt?

---

see https://github.com/kiegroup/drools/pull/2966/files#r448245625